plugins/main/bridge/bridge.go

```diff
@@ -323,30 +355,40 @@ func cmdAdd(args *skel.CmdArgs) error {
 	}
 
 	if n.IsGW {
-		var firstV4Addr net.IP
+		var firstV4GWAddr net.IP
+		var firstV6GWAddr net.IP
 		for _, ipc := range result.IPs {
 			gwn := &net.IPNet{
 				IP:   ipc.Gateway,
 				Mask: ipc.Address.Mask,
 			}
-			if ipc.Gateway.To4() != nil && firstV4Addr == nil {
-				firstV4Addr = ipc.Gateway
-			}
-
-			if err = ensureBridgeAddr(br, gwn, n.ForceAddress); err != nil {
-				return err
+			switch {
+			case ipc.Gateway.To4() != nil && firstV4GWAddr == nil:
+				firstV4GWAddr = ipc.Gateway
+				err = ensureBridgeAddr(br, netlink.FAMILY_V4, gwn, n.ForceAddress)
+				if err != nil {
+					return fmt.Errorf("failed to set bridge addr: %v", err)
+				}
+				if err := ip.EnableIP4Forward(); err != nil {
+					return fmt.Errorf("failed to enable forwarding: %v", err)
+				}
+			case len(ipc.Gateway) == net.IPv6len && firstV6GWAddr == nil:
+				firstV6GWAddr = ipc.Gateway
+				err = ensureBridgeAddr(br, netlink.FAMILY_V6, gwn, n.ForceAddress)
+				if err != nil {
+					return fmt.Errorf("failed to set bridge addr: %v", err)
+				}
+				if err := ip.EnableIP6Forward(); err != nil {
+					return fmt.Errorf("failed to enable forwarding: %v", err)
+				}
```

Comment:
At the moment there are 3 places where it's looping through all IPs in the IPAM results. What I could try to do is combine these 3 loops, and calculate first-V4 / first-V6 pod addresses, default routes, and gateway addresses all in this one loop. If that works, then there would only be one place where were detecting the V4/V6 family. If this is possible, I'll include this in the next uploaded patch.